Dividend name

maxsam4 opened the pull request:
### Please check if the PR fulfills these requirements
- [ ] The commit message follows our Submission guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


### What kind of change does this PR introduce? 
Enhancement


### What is the current behavior? 
Dividends are unnamed


### What is the new behavior?
Dividend have a name


### Does this PR introduce a breaking change?
(What changes might users need to make in their application due to this PR?)
Need to pass Dividend name when creating a Dividend.


### Any Other information:
1) Apparently Sublime added whitespace changes to the whole contracts :/. Add ?w=1 to the URL to see the diff with whitespace ignored (https://github.com/PolymathNetwork/polymath-core/pull/299/commits/213b318984176aedecc31e624c88c13cad913f45?w=1).
2) I made _createDividendWithCheckpointAndExclusions an internal function to save gas when being called from public functions. public -> public function calls require all the parameters to be copied into memory again while public -> internal will just pass on the reference.
3) I am unable to run tests on my local machine for now so using travis for testing.

Fixes #295

maxsam4 committed:
Added name to dividend struct

maxsam4 committed:
test case typos fixed

maxsam4 committed:
Merge branch 'development-1.5.0' into dividend-name

maxsam4 committed:
Updated changelog

adamdossa commented:
@maxsam4 Looks good to me - only comment is please could the function:  
`_CallERC20DividendDepositedEvent`
be renamed to:
`_emitERC20DividendDepositedEvent`

maxsam4 committed:
Renamed a function

Renamed `_CallERC20DividendDepositedEvent` to `_emitERC20DividendDepositedEvent`

maxsam4 committed:
Merge branch 'development-1.5.0' into dividend-name

satyamakgec commented on contracts/modules/Checkpoint/ERC20DividendCheckpoint.sol:
```diff
@@ -1,157 +1,205 @@
+        require(_expiry > _maturity, "Expiry is before maturity");
+        require(_expiry > now, "Expiry is in the past");
+        require(_amount > 0, "No dividend sent");
+        require(_token != address(0), "0x not valid token");
```
add the sanity check on the `_name`. `require(bytes(_name).length > 0);`

maxsam4 replied:
do we need to make _name a required field?

maxsam4 replied:
Changes done

satyamakgec commented on contracts/modules/Checkpoint/EtherDividendCheckpoint.sol:
```diff
@@ -1,147 +1,182 @@
+        require(_excluded.length <= EXCLUDED_ADDRESS_LIMIT, "Too many addresses excluded");
+        require(_expiry > _maturity, "Expiry is before maturity");
+        require(_expiry > now, "Expiry is in the past");
+        require(msg.value > 0, "No dividend sent");
```
same sanity check needed here

maxsam4 replied:
Added

satyamakgec reviewed (changes requested):
I am not a big fan of `bytes32` but if we can replace the datatype to bytes32 for `name` then we can `indexed` it in the events that may useful for the dApp team.

maxsam4 committed:
Merge branch 'development-1.5.0' into dividend-name

maxsam4 committed:
Name changed from string to bytes32

maxsam4 committed:
Removed duplicate test case

pabloruiz55 commented:
looks good! Needs @satyamakgec 's approval
